timely_beliefs/sensors/func_store/knowledge_horizons.py

```diff
@@ -31,6 +34,43 @@ def at_date(
     return event_start - knowledge_time.astimezone(event_start.tzinfo)
 
 
+def x_years_ago_at_date(
+    event_start: datetime | pd.DatetimeIndex,
+    day: int,
+    month: int,
+    x: int = 1,
+    get_bounds: bool = False,
+    z: str | None = None,
+) -> timedelta | pd.TimedeltaIndex | tuple[timedelta, timedelta]:
+    """Compute the sensor's knowledge horizon to represent the event could be known since some date, `x` years ago.
+
+    For example, it can be used for a tax rate that changes annually and with a known publication date.
+
+    :param event_start:     Start of the event, used as an anchor for determining the knowledge horizon.
+    :param day:             Reference day of the month of the annual date to compare against.
+    :param month:           The month of the annual date to compare against.
+    :param x:               The number of years to shift the reference date to.
+    :param z:           Timezone string.
+    :param get_bounds:      If True, this function returns bounds on the possible return value.
+                            These bounds are normally useful for creating more efficient database queries when filtering by belief time.
+    """
+
+    if x < 0:
+        raise ValueError("Negative value for `x` not supported.")
+
+    if get_bounds:
+        return timedelta(days=(x - 1) * 366 - 3), timedelta(days=(x + 1) * 366 + 1)
```

Comment:
Please add a comment for future reference, explaining the numbers used, similar to the inline comment for the bounds in `x_days_ago_at_y_oclock`.

Reply:
I've added a couple of comments + made the 366 into a constant variable.